CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -84,10 +86,10 @@ public async Task<Lesson> AssignMentorToLessonAsync(AssignMentorToLessonDto ids)
 
             await _unitOfWork.CommitAsync();
 
-            return foundLesson;
+            return Result<Lesson>.GetSuccess(foundLesson);
         }
 
-        public async Task<LessonDto> UpdateLessonAsync(long id, UpdateLessonDto lessonModel)
+        public async Task<Result<LessonDto>> UpdateLessonAsync(long id, UpdateLessonDto lessonModel)
```

Comment:
Don't forget to call .ToActionResult in controller action. Otherwise, it will change schema of response body.